zebra/debug_nl.c

```diff
@@ -0,0 +1,1291 @@
+	FLAG_WRITE(IFF_AUTOMEDIA, "AUTOMEDIA");
+	FLAG_WRITE(IFF_DYNAMIC, "DYNAMIC");
+
+	return (bufp);
+}
+
+const char *rtm_flags2str(uint32_t flags, char *buf, size_t buflen)
+{
+	const char *bufp = buf;
+	int first = 1;
+	int wlen;
+
+	*buf = 0;
+	FLAG_WRITE(RTM_F_NOTIFY, "NOTIFY");
+	FLAG_WRITE(RTM_F_CLONED, "CLONED");
+	FLAG_WRITE(RTM_F_EQUALIZE, "EQUALIZE");
+
+	return (bufp);
+}
+
+const char *neigh_state2str(uint32_t flags, char *buf, size_t buflen)
+{
+	const char *bufp = buf;
+	int first = 1;
+	int wlen;
+
+	*buf = 0;
+	FLAG_WRITE(NUD_INCOMPLETE, "INCOMPLETE");
+	FLAG_WRITE(NUD_REACHABLE, "REACHABLE");
+	FLAG_WRITE(NUD_STALE, "STALE");
+	FLAG_WRITE(NUD_DELAY, "DELAY");
+	FLAG_WRITE(NUD_PROBE, "PROBE");
+	FLAG_WRITE(NUD_FAILED, "FAILED");
+	FLAG_WRITE(NUD_NOARP, "NOARP");
+	FLAG_WRITE(NUD_PERMANENT, "PERMANENT");
+
+	return (bufp);
+}
+
+const char *neigh_flags2str(uint32_t flags, char *buf, size_t buflen)
+{
+	const char *bufp = buf;
+	int first = 1;
+	int wlen;
+
+	*buf = 0;
+	FLAG_WRITE(NTF_USE, "USE");
+	FLAG_WRITE(NTF_SELF, "SELF");
+	FLAG_WRITE(NTF_MASTER, "MASTER");
+	FLAG_WRITE(NTF_PROXY, "PROXY");
+	FLAG_WRITE(NTF_EXT_LEARNED, "EXT_LEARNED");
+#ifdef NTF_OFFLOADED
+	FLAG_WRITE(NTF_OFFLOADED, "OFFLOADED");
+#endif /* NTF_OFFLOADED */
+	FLAG_WRITE(NTF_ROUTER, "ROUTER");
+
+	return (bufp);
+}
+
+const char *ifa_flags2str(uint32_t flags, char *buf, size_t buflen)
+{
+	const char *bufp = buf;
+	int first = 1;
+	int wlen;
+
+	*buf = 0;
+	FLAG_WRITE(IFA_F_SECONDARY, "SECONDARY");
+	FLAG_WRITE(IFA_F_NODAD, "NODAD");
+	FLAG_WRITE(IFA_F_OPTIMISTIC, "OPTIMISTIC");
+	FLAG_WRITE(IFA_F_DADFAILED, "DADFAILED");
+	FLAG_WRITE(IFA_F_HOMEADDRESS, "HOMEADDRESS");
+	FLAG_WRITE(IFA_F_DEPRECATED, "DEPRECATED");
+	FLAG_WRITE(IFA_F_TENTATIVE, "TENTATIVE");
+	FLAG_WRITE(IFA_F_PERMANENT, "PERMANENT");
+	FLAG_WRITE(IFA_F_MANAGETEMPADDR, "MANAGETEMPADDR");
+	FLAG_WRITE(IFA_F_NOPREFIXROUTE, "NOPREFIXROUTE");
+	FLAG_WRITE(IFA_F_MCAUTOJOIN, "MCAUTOJOIN");
+	FLAG_WRITE(IFA_F_STABLE_PRIVACY, "STABLE_PRIVACY");
+
+	return (bufp);
+}
+
+const char *nh_flags2str(uint32_t flags, char *buf, size_t buflen)
+{
+	const char *bufp = buf;
+	int first = 1;
+	int wlen;
+
+	*buf = 0;
+	FLAG_WRITE(RTNH_F_DEAD, "DEAD");
+	FLAG_WRITE(RTNH_F_PERVASIVE, "PERVASIVE");
+	FLAG_WRITE(RTNH_F_ONLINK, "ONLINK");
+	FLAG_WRITE(RTNH_F_OFFLOAD, "OFFLOAD");
+	FLAG_WRITE(RTNH_F_LINKDOWN, "LINKDOWN");
+	FLAG_WRITE(RTNH_F_UNRESOLVED, "UNRESOLVED");
+
+	return (bufp);
+}
+
+/*
+ * Netlink abstractions.
+ */
+static void nllink_linkinfo_dump(struct rtattr *rta, size_t msglen)
+{
+	size_t plen;
+	char dbuf[128];
+
+next_rta:
+	/* Check the header for valid length and for outbound access. */
+	if (RTA_OK(rta, msglen) == 0)
+		return;
+
+	plen = RTA_PAYLOAD(rta);
+	zlog_debug("      linkinfo [len=%d (payload=%lu) type=(%d) %s]",
+		   rta->rta_len, plen, rta->rta_type,
+		   rta_type2str(rta->rta_type));
+	switch (rta->rta_type) {
+	case IFLA_INFO_KIND:
+		if (plen < sizeof(char)) {
```

Comment:
`sizeof(char)` is defined by the standard to always be 1, it's probably clearer to put that or `plen == 0`

Reply:
Fixed.